### PR TITLE
Ffe-file-upload redesign

### PR DESCRIPTION
**Pull request by jafeld**
Big redesign of ffe-file-upload, new design can be seen here (requires access): https://www.figma.com/file/cn3VFs107XnyxeLo4Du4ny/Fil-Opplasting---Master?node-id=0%3A1

Some examples of the new design: 

ffe-file-upload-1

ffe-file-upload-2

These are breaking changes.

### CODE REVIEW

File: packages/ffe-file-upload-react/src/FileItem.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        {// File is loading
+        !props.file.document && !props.file.error && (
+            <div className="ffe-file-upload__info-section__stencil">
+                <div className="ffe-file-upload__info-section__stencil__info">
+                    <div className="ffe-file-upload__info-section__stencil__info__background" />
+                    <div className="ffe-file-upload__info-section__stencil__info__text">
+                        {props.file.name}
+                    </div>
+                </div>
+                <button
```
Comment (ivarni) [inline packages/ffe-file-upload-react/src/FileItem.js:15]: Could you throw a `type="button"` on this so it doesn't trigger a submit if it's inside a `form`?

File: packages/ffe-file-upload-react/src/FileItem.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                        id={props.file.name}
+                        className="ffe-file-upload__info-section__delete__icon"
+                    />
+                </button>
+            </div>
+        )}
+        {// File finished loading
+        props.file.document && (
+            <div className="ffe-file-upload__info-section__loaded">
+                <button
```
Comment (ivarni) [inline packages/ffe-file-upload-react/src/FileItem.js:32]: Same with this button, and any other `button` elements

File: packages/ffe-file-upload-react/src/FileUpload.js [hunk 3]
```diff
@@ -29,97 +33,106 @@ class FileUpload extends React.Component {
...
+                            </div>
+                            <div className="ffe-file-upload__info-section__subtext">
+                                {infoSubText}
+                            </div>
+                        </>
+                    )}
+                </div>
+                <div
+                    className="ffe-file-upload__upload-section"
+                    role="presentation"
```
Comment (ivarni) [inline packages/ffe-file-upload-react/src/FileUpload.js:95]: `role="presentation"` will remove the semantic meaning of the element and all child elements, I'm not sure that's needed here. How assistive tech handles it varies as far as I know but I don't think it helps in this context.

**Comment by ivarni** [comment]
It would be great if you could squash the commits since lerna will release a new version for each one. It's also good to include the text `BREAKING CHANGE` in the commit message as that will make lerna release this as a new major so noone accidentally gets API changes when updating minor/fix version.

I'm not sure if you need separate commits for the CSS and React packages or not.

**Comment by antidecaf** [comment]
You do need separate commits for CSS and React packages, as they are published to npm separately.

File: packages/ffe-file-upload-react/src/FileUpload.js [hunk 4]
```diff
@@ -29,97 +33,107 @@ class FileUpload extends React.Component {
...
+                        this.setState({ hover: true });
+                    }}
+                    onDragLeave={() => this.setState({ hover: false })}
                 >
-                    <span className="ffe-file-upload__button__label-icon" />
-                    <span id={`${id}-label`}>{label}</span>
-                </button>
+                    <div
+                        className={`ffe-file-upload__upload-section__border${
+                            this.state.hover
```
Comment (antidecaf) [inline packages/ffe-file-upload-react/src/FileUpload.js:105]: Rather than toggling the class, styling should be applied to the `:hover` state in CSS/Less.

Comment (jafeld) [reply packages/ffe-file-upload-react/src/FileUpload.js:105]: This won't work, as it's only supposed to trigger styling when dragging files over the object, just mousing over should not trigger it

File: packages/ffe-file-upload/less/file-upload.less [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+@ffe-blue-royal-fill: @ffe-blue-royal-escape;
+@ffe-blue-azure-escape: escape('@{ffe-blue-azure}');
+@ffe-blue-azure-fill: @ffe-blue-azure-escape;
+
+.ffe-file-upload {
+    margin-top: 45px;
+    display: block;
+
+    @media (min-width: @breakpoint-md) {
+        display: -webkit-box;
```
Comment (antidecaf) [inline packages/ffe-file-upload/less/file-upload.less:11]: Vendor prefixes are not necessary to include, as they will be generated by autoprefixer.

File: README.md [hunk 6]
```diff
@@ -59,5 +58,5 @@ bidrag. Ta en titt på [CONTRIBUTING.md](CONTRIBUTING.md) for hvordan du går fr
 
 ## Licenses
 
-* Source code is licensed under MIT
-* The MuseoSans fonts are licensed separately. See LICENSE-fonts.md.
+-   Source code is licensed under MIT
```
Comment (antidecaf) [inline README.md:61]: Not sure if this is Prettier running wild, but this change breaks the markdown generated bullet list. The reformatting of this file probably shouldn't be included in the final commit.

Comment (jafeld) [reply README.md:61]: It is, I will fix this once I squash the commits and fix the changes to files that shouldn't have been included :)

File: packages/ffe-datepicker-react/src/datepicker/Datepicker.js [hunk 7]
```diff
@@ -309,9 +309,9 @@ export default class Datepicker extends Component {
         const latestValue = validateDate(value) ? value : lastValidDate;
 
         if (this.state.ariaInvalid && !inputProps['aria-describedby']) {
-            inputProps[
-                'aria-describedby'
-            ] = `date-input-validation-${this.datepickerId}`;
+            inputProps['aria-describedby'] = `date-input-validation-${
```
Comment (antidecaf) [inline packages/ffe-datepicker-react/src/datepicker/Datepicker.js:312]: This looks like a Prettier related accidental reformatting too.

File: packages/ffe-searchable-dropdown-react/src/ListContainer.js [hunk 8]
```diff
@@ -28,8 +28,8 @@ const ListContainer = ({
                 ? dropdownList.slice(0, maxRenderedDropdownElements)
                 : dropdownList
             : typeof noMatch === 'string'
-                ? []
-                : noMatch.items;
+            ? []
```
Comment (antidecaf) [inline packages/ffe-searchable-dropdown-react/src/ListContainer.js:31]: Another apparent Prettier related reformatting

File: packages/ffe-file-upload/less/file-upload.less [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+
+        @media (min-width: @breakpoint-md) {
+            margin-right: 20px;
+            max-width: 100%;
+            -webkit-box-flex: 0;
+            -ms-flex: 0px;
+            flex: 0px;
+        }
+
+        &__title {
```
Comment (antidecaf) [inline packages/ffe-file-upload/less/file-upload.less:31]: This goes for much of the styling structure - Multi level nesting of elements should be avoided, and the class names should not reflect the dom structure. In stead, each element in a block should be independent (but they can still be nested in the markup). See [Why not create elements of elements?](https://en.bem.info/methodology/faq/#why-not-create-elements-of-elements-block__elem1__elem2) and [BEM/Nesting](https://en.bem.info/methodology/quick-start/#nesting-1)

Comment (jafeld) [reply packages/ffe-file-upload/less/file-upload.less:31]: Will rewrite this when possible, didn't have much time to spare before :)

**Comment by jafeld** [comment]
new PR: https://github.com/SpareBank1/designsystem/pull/694
closing this one

**Pull request closed without merging**